### DISCUSSION TITLE
JSON schema 的 ArrayItems ，新增的每一行中怎么拿到index呢？

### DISCUSSION THREAD

**Discussion by legcc** [alibaba/formily#2711] [Q&A 2.x] [answered]
```
const schema = {
  type: 'object',
  properties: {
      array: {
      type: 'array',
      'x-component': 'ArrayItems',
      'x-decorator': 'FormItem',
      title: '对象数组',
      items: {
        type: 'object',
        properties: {
          space: {
            type: 'void',
            'x-component': 'Space',
            properties: {
              // 我需要在第一行隐藏input字段
              input: {
                type: 'string',
                title: '输入框',
                'x-decorator': 'FormItem',
                'x-component': 'Input',
              },
              select: {
                type: 'string',
                title: '下拉框',
                enum: [
                  { label: '选项1', value: 1 },
                  { label: '选项2', value: 2 },
                ],
                'x-decorator': 'FormItem',
                'x-component': 'Select',
                'x-component-props': {
                  style: {
                    width: 160,
                  },
                },
              },
              remove: {
                type: 'void',
                'x-decorator': 'FormItem',
                'x-component': 'ArrayItems.Remove',
              },
            },
          },
        },
      },
      properties: {
        add: {
          type: 'void',
          title: '添加条目',
          'x-component': 'ArrayItems.Addition',
        },
      },
    },
  ｝
｝
```

**Comment by legcc** [comment]
> ```
> const schema = {
>   type: 'object',
>   properties: {
>       array: {
>       type: 'array',
>       'x-component': 'ArrayItems',
>       'x-decorator': 'FormItem',
>       title: '对象数组',
>       items: {
>         type: 'object',
>         properties: {
>           space: {
>             type: 'void',
>             'x-component': 'Space',
>             properties: {
>               // 我需要在第一行隐藏input字段
>               input: {
>                 type: 'string',
>                 title: '输入框',
>                 'x-decorator': 'FormItem',
>                 'x-component': 'Input',
>               },
>               select: {
>                 type: 'string',
>                 title: '下拉框',
>                 enum: [
>                   { label: '选项1', value: 1 },
>                   { label: '选项2', value: 2 },
>                 ],
>                 'x-decorator': 'FormItem',
>                 'x-component': 'Select',
>                 'x-component-props': {
>                   style: {
>                     width: 160,
>                   },
>                 },
>               },
>               remove: {
>                 type: 'void',
>                 'x-decorator': 'FormItem',
>                 'x-component': 'ArrayItems.Remove',
>               },
>             },
>           },
>         },
>       },
>       properties: {
>         add: {
>           type: 'void',
>           title: '添加条目',
>           'x-component': 'ArrayItems.Addition',
>         },
>       },
>     },
>   ｝
> ｝
> ```

想在items第一行隐藏input字段，但是在input上通过x-reactions没办法拿到当前行的索引，看到有一个ArrayItems.useIndex hook 试了一下没有效果，不知道是不是使用方式不对，没看到案例，麻烦老师看看😬

**Comment by janryWang** [comment, marked as answer]
自己写自定义组件，在自定义组件里控制第一行输入框不显示

**Reply by qreal829475** [reply]
请问怎么能够在子组件中，获得索引呢？

**Reply by ChenAlmost** [reply]
<img width="833" alt="image" src="https://user-images.githubusercontent.com/31819795/172187544-35af9529-5c8a-4e76-940d-2ea1102969ef.png">
我这样可以拿到index，可以参考一下，看可行不

**Reply by feynmx** [reply]
也可试下$index获取索引

**Reply by ChenAlmost** [reply]
> 也可试下$index获取索引

完美解决！
